### ISSUE TITLE
Ads per hour setting not retained on upgrade to 1.41.x

### ISSUE THREAD

**Issue by LaurenWags** [brave/brave-browser#23383]
## Description 

Ads per hour setting is not retained on upgrade from 1.39.x --> 1.41.x, it is being overwritten by the value provided by variations server.

## Steps to Reproduce

   1. Set up a profile with 1.39.120 (current release version).
   2. Enable rewards, change Ads per hour setting from default to some other value that is not 5 or 10 (I used 1).
   3. Close 1.39.120.
   4. Rename profile to "upgrade" to 1.41.x (current Nightly version)
   5. Launch 1.41.x
   6. Look at Ads per hour setting. It is now at 10.

## Actual result:

<img width="1349" alt="1 41 x" src="https://user-images.githubusercontent.com/28145373/172949634-77898f11-62d9-4980-b94b-17cc6108185d.png">

## Expected result:

<img width="1354" alt="1 39 x" src="https://user-images.githubusercontent.com/28145373/172949596-2af978c8-9d49-434a-9899-e40e5c1dbabd.png">

## Reproduces how often: 

easily

## Brave version (brave://version info)

Brave | 1.41.50 Chromium: 102.0.5005.99 (Official Build) nightly (x86_64)
-- | --
Revision | 870f7bcc58dfa811cc68c2186439721385e086d0-refs/branch-heads/5005@{#1125}
OS | macOS Version 12.4 (Build 21F79)

## Version/Channel Information:

- Can you reproduce this issue with the current release? 1.39.x --> 1.41.x yes (reproduced by @btlechowski @kjozwiak )
- Can you reproduce this issue with the beta channel? 1.40.x --> 1.41.x yes (reproduced by @btlechowski )
- Can you reproduce this issue with the nightly channel? 1.41.x --> 1.41.x TBD

## Other Additional Information:

- Does the issue resolve itself when disabling Brave Shields? 
- Does the issue resolve itself when disabling Brave Rewards? 
- Is the issue reproducible on the latest version of Chrome? 

## Miscellaneous Information:

cc @tmancey @jsecretan

**Comment by LaurenWags** [comment]
Per additional discussion via slack, issue likely is a result of https://github.com/brave/brave-core/pull/13675 which was included in 1.41.47.

Confirmed issue does not reproduce on upgrade to 1.41.46, but does reproduce when upgrading to 1.41.47.

Thanks for the help @tmancey @rebron

**Comment by stephendonner** [comment]
## Verified `PASSED` using 

Brave | 1.41.77 Chromium: 103.0.5060.53 (Official Build) beta (64-bit) 
--|--
Revision | a1711811edd74ff1cf2150f36ffa3b0dae40b17f-refs/branch-heads/5060@{#853}
OS | Linux

Verified STR (using current release version of `1.40.105` instead of 1.39.x):
* Confirmed user selected ads per hour value was retained on upgrade
* Closed/relaunched 1.41.x several times and confirmed user selected ads per hour value was not changed

1.40.105 | 1.41.77
----- | -----
<img width="1048" alt="Screen Shot 2022-06-23 at 4 03 57 PM" src="https://user-images.githubusercontent.com/387249/175429370-aba6446b-599f-4f30-a577-2fb0c783306b.png"> | <img width="1048" alt="Screen Shot 2022-06-23 at 4 04 56 PM" src="https://user-images.githubusercontent.com/387249/175429373-23e7ad52-8046-4958-bf6c-12cf67cb1adb.png">

Also verified the following:
* Clean install of 1.41.77, enable Rewards, relaunch browser. Confirmed for (production env), the default value of 10 ads per hour was set via the server.
* Changed this value to something else (I used 5)
* Close and relaunch, user selected value was retained

**Comment by GeetaSarvadnya** [comment]
Verification is in-progress

```

Brave | 1.41.93 Chromium: 103.0.5060.114 (Official Build) (64-bit)
-- | --
Revision | a1c2360c5b02a6d4d6ab33796ad8a268a6128226-refs/branch-heads/5060@{#1124}
OS | Windows 10 Version 21H2 (Build 19044.1766)

```
